docs/how-to/create-chiselled-docker-image.md

```diff
@@ -0,0 +1,181 @@
+is able to run Python 3 scripts and commands.
+
+Goals:
+
+- The [image entrypoint] must be `python3`.
+- The image must be chiselled i.e. there must not be anything else.
+- The image must be able to run Python 3 scripts.
+- The image must be minimal.
+
+To build this image, we will use [multi-stage docker builds].
+
+In earlier stage(s),
+
+1. We will install Chisel and other dependencies that we may
+need.
+2. We will install the slices using Chisel and prepare a root file system.
+
+Finally, we will copy the root file system to the `/` directory in the final
+stage. The final stage base image will be [`scratch`], i.e. nothing. Thus, the
+image only contains the root file system installed by Chisel and nothing else.
+
+
+## Install Chisel and dependencies
+
+To make building easier and configurable, let's first define some variables that
+we can re-use.
+
+```docker
+ARG UBUNTU_RELEASE=24.04
+ARG USER=app UID=101 GROUP=app GID=101
+ARG CHISEL_VERSION=1.1.0
+```
+
+Let's now initialize a new build stage where we install Chisel and dependencies
+and prepare the root file system.
+
+```docker
+FROM ubuntu:$UBUNTU_RELEASE AS builder
+ARG USER UID GROUP GID TARGETARCH UBUNTU_RELEASE CHISEL_VERSION
+SHELL ["/bin/bash", "-oeux", "pipefail", "-c"]
+```
+
+We will need the `ca-certificates` package to fetch files. Let's install that.
+
+```docker
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y ca-certificates \
+    && apt-get clean -y \
+    && rm -rf /var/lib/apt/lists/*
+```
+
+We can now install Chisel. Make sure to install the latest version. Visit the
+{{latest_release_page}} to determine the latest version and override
+`CHISEL_VERSION` appropriately.
+
+```docker
+ADD "https://github.com/canonical/chisel/releases/download/v${CHISEL_VERSION}/chisel_v${CHISEL_VERSION}_linux_${TARGETARCH}.tar.gz" \
+    chisel.tar.gz
+RUN tar -xvf chisel.tar.gz -C /usr/bin/
+```
+
+
+## Prepare chiselled root file system
+
+Now that we have Chisel installed in the `builder` stage, we can install the
+appropriate slices and prepare the root file system.
+
+The [`python3_standard` slice] provides `python3`. We will install this slice,
+along with a few others:
+
+- `base-files_base` for file structure.
+- `base-files_release-info` for release info.
+- `base-files_chisel` for the chisel manifest.
+- `ca-certificates_data` for network support.
+
+```docker
+RUN mkdir /rootfs \
+    && chisel cut --release "ubuntu-$UBUNTU_RELEASE" --root /rootfs \
+        base-files_base \
+        base-files_release-info \
+        base-files_chisel \
+        ca-certificates_data \
+        python3_standard
+```
+
+We will also install two users in the root file system - `USER` and `root`.
+
+```docker
+RUN install -d -m 0755 -o "$UID" -g "$GID" "/rootfs/home/$USER" \
+    && echo -e "root:x:0:\n$GROUP:x:$GID:" >/rootfs/etc/group \
+    && echo -e "root:x:0:0:root:/root:/noshell\n$USER:x:$UID:$GID::/home/$USER:/noshell" >/rootfs/etc/passwd
+```
```

Comment:
why not use the `base-passwd` slices?

Reply:
i'd also be ok with jsut copying the /etc/passwd from the ubuntu builder stage, since it already has the `ubuntu:1000` user that we can use

Reply:
Used `base-passwd`, thanks!